lang/cs/Org.Apache.REEF.Examples/MachineLearning/KMeans/KMeansDriverHandlers.cs

```diff
@@ -79,15 +80,23 @@ public KMeansDriverHandlers()
                 _executionDirectory); 
 
             _totalEvaluators = _partitionsNumber + 1;
-            _mpiDriver = new MpiDriver(Identifier, Constants.MasterTaskId, _fanOut, new AvroConfigurationSerializer());
-
-            _commGroup = _mpiDriver.NewCommunicationGroup(
-               Constants.KMeansCommunicationGroupName,
-               _totalEvaluators)
-                   .AddBroadcast(Constants.CentroidsBroadcastOperatorName, new BroadcastOperatorSpec<Centroids>(Constants.MasterTaskId, new CentroidsCodec()))
-                   .AddBroadcast(Constants.ControlMessageBroadcastOperatorName, new BroadcastOperatorSpec<ControlMessage>(Constants.MasterTaskId, new ControlMessageCodec()))
-                   .AddReduce(Constants.MeansReduceOperatorName, new ReduceOperatorSpec<ProcessedResults>(Constants.MasterTaskId, new ProcessedResultsCodec(), new KMeansMasterTask.AggregateMeans()))
+
+            IConfiguration mpiDriverConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindStringNamedParam<MpiConfigurationOptions.DriverId>(Identifier)
+                .BindStringNamedParam<MpiConfigurationOptions.MasterTaskId>(Constants.MasterTaskId)
+                .BindStringNamedParam<MpiConfigurationOptions.GroupName>(Constants.KMeansCommunicationGroupName)
+                .BindIntNamedParam<MpiConfigurationOptions.FanOut>(_fanOut.ToString(CultureInfo.InvariantCulture).ToString(CultureInfo.InvariantCulture))
```

Comment:
we should not need to specify the FanOut option now in user code right? since it now has a default value of 2. It still feels a bit unnatural to have it appears in user code.